internal/services/cart_service.go

```diff
@@ -0,0 +1,110 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"innogocartapi/internal/models"
+)
+
+var ErrServer error = errors.New("internal server error")
```

Comment:
Service doesn't know about such a thing as a "server"
```suggestion
var ErrInternal error = errors.New("internal server error")
```